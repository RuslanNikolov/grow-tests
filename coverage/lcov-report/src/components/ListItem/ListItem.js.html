<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/ListItem/ListItem.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/ListItem</a> ListItem.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from "react";
&nbsp;
const ListItem = props =&gt; {
  const { title, image, url } = props;
&nbsp;
  return (
    &lt;div className="listItem"&gt;
      {image ? (
        &lt;div
          style={{ backgroundImage: `url(${image})` }}
          className="listItem__image"
        /&gt;
      ) : (
        &lt;span className="noImage" /&gt;
      )}
      &lt;div className="listItem__info"&gt;
        &lt;span className="listItem__title"&gt;{title}&lt;/span&gt;
        &lt;a className="listItem__Link" href={url} target="_blank"&gt;
          Link
        &lt;/a&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  );
};
&nbsp;
export default ListItem;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Sep 15 2019 21:43:50 GMT+0300 (Eastern European Summer Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
