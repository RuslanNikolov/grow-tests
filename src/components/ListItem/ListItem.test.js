import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ListItem from './ListItem'

const mockItem1 = {
    id: 1,
    title: 'title1',
    url: 'url1',
    image: 'thumbnailUrl1'
}

const mockItem2 = {
    id: 1,
    title: 'title1',
    url: 'url1',
}

describe('ListItem', () => {
    let wrapper;

    it('should render the UI with image', () => {
        wrapper = shallow(<ListItem {...mockItem1} />)

        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('should render the UI without image', () => {
        wrapper = shallow(<ListItem {...mockItem2} />)

        expect(toJson(wrapper)).toMatchSnapshot()
    })

})