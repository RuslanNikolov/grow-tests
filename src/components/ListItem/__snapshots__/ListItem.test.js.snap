// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render the UI with image 1`] = `
<div
  className="listItem"
>
  <div
    className="listItem__image"
    style={
      Object {
        "backgroundImage": "url(thumbnailUrl1)",
      }
    }
  />
  <div
    className="listItem__info"
  >
    <span
      className="listItem__title"
    >
      title1
    </span>
    <a
      className="listItem__Link"
      href="url1"
      target="_blank"
    >
      Link
    </a>
  </div>
</div>
`;

exports[`ListItem should render the UI without image 1`] = `
<div
  className="listItem"
>
  <span
    className="noImage"
  />
  <div
    className="listItem__info"
  >
    <span
      className="listItem__title"
    >
      title1
    </span>
    <a
      className="listItem__Link"
      href="url1"
      target="_blank"
    >
      Link
    </a>
  </div>
</div>
`;
